tests/files/type/test_type_base.py

```diff
@@ -0,0 +1,34 @@
+import pathlib
+
+import pytest
+
+from astro.constants import FileType
+from astro.files.type import get_filetype
+
+sample_file = pathlib.Path(pathlib.Path(__file__).parent.parent, "data/sample.csv")
+sample_filepaths_per_filetype = [
+    (FileType.CSV, "sample.csv"),
+    (FileType.JSON, "sample.json"),
+    (FileType.NDJSON, "sample.ndjson"),
+    (FileType.PARQUET, "sample.parquet"),
```

Comment:
I'd love to see this type of thing unified in a fixture or using another strategy - so we don't need to redefine it per test/per test file.

Reply:
added a fixture - type_method_map_fixture